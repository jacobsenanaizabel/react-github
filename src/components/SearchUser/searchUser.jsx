import React from 'react'
import { Link } from "react-router-dom"
import axios from 'axios'

import './searchUser.scss'

export default class SearchUser extends React.Component{

    constructor(props) {
        super(props);
        this.state = {
            userListReturn : [],
            totalUsers : 0,
            userList : this.props.location.state.user
        }
    }

    componentDidMount() {
        axios.get(`https://api.github.com/search/users?q=`+ this.state.userList, { crossdomain: true } )
            .then(response => {
                this.setState({totalUsers : response.data.total_count});
                this.setState({userListReturn : response.data.items});
            }
        )
    }

    render() {
        return(     
            <div className="user-list-container">
                <h2 className="title">Resultados achados {this.state.totalUsers}!</h2> 
                { this.state.userListReturn.map(user => 
                    <div className="card text-center">
                        <div className="card-header">
                            {user.login}
                        </div>
                        <div className="card-body">
                            <img  src={user.avatar_url} class="card-avatar"/>
                            <h4 className="card-title">Score do usuario</h4>
                            <p className="card-text">{user.score}</p>
                            <a className="btn btn-primary">
                            <Link to={{pathname: '/user',
                                       state: { profile: user.url } }}> Saiba mais </Link></a>
                        </div>
                        <div className="card-footer text-muted">
                            <label> ID Do Usuario: {user.id}</label>
                        </div>
                    </div>
                )}
            </div>
        )
    }
}