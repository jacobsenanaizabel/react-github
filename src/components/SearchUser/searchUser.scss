.user-list-container {
    align-content: space-around;
    align-items: center;
    display: flex;
    flex-direction: row;
    flex-flow: row wrap;
    font-family: 'Cabin Sketch';
    justify-content: center;
    position: relative;

    .button-more{
        margin-bottom: 5px;
    }

    .card.text-center {
        margin-bottom: 2%;
        flex-basis: 37%;
        margin-left: 3%;

        .card-avatar{
            width: 23%;
            margin-top: 3%;
        }
    }

    .title {
        color: #363636;
        font-size: 58px;
        line-height: 1.125;
        width: 100%;
        margin-left: 30%;
        margin-bottom: 3%;
    }
}
