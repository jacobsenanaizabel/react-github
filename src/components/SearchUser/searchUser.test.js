import React from 'react'
import { shallow, mount } from 'enzyme'
import SearchUser from './searchUser'

describe('Search User', () =>{
    it('should be return list of users', () =>{
        const wrapper = shallow(<SearchUser/>)

        const userList = wrapper.state().userListReturn
        expect(userList).toBeDefined
    })

})