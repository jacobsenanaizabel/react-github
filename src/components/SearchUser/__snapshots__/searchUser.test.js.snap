// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchUser /> renders repository snapshot 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "canGo": [Function],
      "createHref": [Function],
      "entries": Array [
        Object {
          "hash": "",
          "key": "d50kq3",
          "pathname": "/search",
          "search": "",
          "state": undefined,
        },
        Object {
          "hash": "",
          "key": "vg8sgo",
          "pathname": "/",
          "search": "",
          "user": "jacobsenanaizabel",
        },
      ],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "index": 0,
      "length": 2,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "key": "d50kq3",
        "pathname": "/search",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <SearchUser />
</Router>
`;
